src/main/java/de/unijena/cheminf/fragment/fingerprint/FragmentFingerprinter.java

```diff
@@ -153,23 +137,106 @@ public FragmentFingerprinter(List<String> aFragmentList) throws NullPointerExcep
      */
     @Override
     public IBitFingerprint getBitFingerprint(List<String> aListOfUniqueSmiles) throws NullPointerException, IllegalArgumentException {
-        this.validityCheckOfParameterList(aListOfUniqueSmiles,"aListOfUniqueSmiles (list of string instances) is null.",
-                "aListOfUniqueSmiles (at least one list element) is null.",
-                "aListOfUniqueSmiles (at least one list element) is blank/empty.");
+        this.validityCheckOfParameterList(aListOfUniqueSmiles,"Given list of string instances is null.",
+                "Given list includes at least one null element.",
+                "Given list includes at least one blank/empty element.");
+        if (this.cacheFingerprintSetting && this.cacheBitSetFingerprint != null) {
+            return this.cacheBitSetFingerprint;
```

Comment:
On second thought, please simply remove all the caching. Generating the FP is not that expensive. Always checking whether the same fragment set is given will be more costly in total.